tests/topotests/zebra_rib/test_zebra_import.py

```diff
@@ -0,0 +1,282 @@
+        """
+    )
+
+    expected = json.loads(open(initial_json_file).read())
+    test_func = partial(
+        topotest.router_json_cmp, r1, "show ip route json", expected
+    )
+    _, result = topotest.run_and_expect(test_func, None, count=5, wait=1)
+    assert result is None, '"r1" JSON output mismatches'
+
+    step("Re-import with distance and verify correct distance")
+    r1.vtysh_cmd(
+        """
+        conf term
+         ip import-table 10 distance 123
+        """)
+    
+    import_json_file = "{}/r1/import_table_4.json".format(CWD)
+    expected = json.loads(open(import_json_file).read())
+    test_func = partial(
+        topotest.router_json_cmp, r1, "show ip route json", expected
+    )
+    _, result = topotest.run_and_expect(test_func, None, count=5, wait=1)
+    assert result is None, '"r1" JSON output mismatches'
+
+def test_zebra_mrib_import(request):
+    "Verify router starts with the initial MRIB"
+    tgen = get_topogen()
+    tc_name = request.node.name
+    write_test_header(tc_name)
+
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+    r1 = tgen.gears["r1"]
+
+    step("Verify initial main MRIB routing table")
+    initial_json_file = "{}/r1/import_init_mrib_table.json".format(CWD)
+    expected = json.loads(open(initial_json_file).read())
+    test_func = partial(
+        topotest.router_json_cmp, r1, "show ip rpf json", expected
+    )
+    _, result = topotest.run_and_expect(test_func, None, count=5, wait=1)
+    assert result is None, '"r1" JSON output mismatches'
+
+    r1.vtysh_cmd(
+        """
+        conf term
+         ip import-table 10 mrib
+        """)
+    
+    import_json_file = "{}/r1/import_mrib_table_2.json".format(CWD)
+    expected = json.loads(open(import_json_file).read())
+    test_func = partial(
+        topotest.router_json_cmp, r1, "show ip rpf json", expected
+    )
+    _, result = topotest.run_and_expect(test_func, None, count=5, wait=1)
+    assert result is None, '"r1" JSON output mismatches'
+
+    step("Add a new static route and verify it gets added")
+    r1.vtysh_cmd(
+        """
+        conf term
+         ip route 10.20.0.0/24 10.10.0.2 table 10
+        """
+    )
+
+    sync_json_file = "{}/r1/import_mrib_table_3.json".format(CWD)
+    expected = json.loads(open(sync_json_file).read())
+    test_func = partial(
+        topotest.router_json_cmp, r1, "show ip rpf json", expected
+    )
+    _, result = topotest.run_and_expect(test_func, None, count=5, wait=1)
+    assert result is None, '"r1" JSON output mismatches'
+
+    step("Remove the static route and verify it gets removed")
+    r1.vtysh_cmd(
+        """
+        conf term
+         no ip route 10.20.0.0/24 10.10.0.2 table 10
+        """
+    )
+
+    expected = json.loads(open(import_json_file).read())
+    test_func = partial(
+        topotest.router_json_cmp, r1, "show ip rpf json", expected
+    )
+    _, result = topotest.run_and_expect(test_func, None, count=5, wait=1)
```

Comment:
Please use higher values of count/wait. Otherwise we get a warning which should be fixed eventually.